pkgs/dart_mcp/lib/src/api/tools.dart

```diff
@@ -1081,6 +1094,56 @@ extension type const StringSchema.fromMap(Map<String, Object?> _value)
   }
 }
 
+/// A JSON Schema definition for a set of allowed string values.
+extension type EnumSchema.fromMap(Map<String, Object?> _value)
+    implements Schema {
+  factory EnumSchema({
+    String? title,
+    String? description,
+    required Set<String> values,
+  }) => EnumSchema.fromMap({
+    'type': JsonType.enumeration.typeName,
+    if (title != null) 'title': title,
+    if (description != null) 'description': description,
+    'enum': values.toSet(),
+  });
+
+  /// A title for this schema, should be short.
+  String? get title => _value['title'] as String?;
+
+  /// A description of this schema.
+  String? get description => _value['description'] as String?;
+
+  /// The allowed enum values.
+  Set<String> get values => (_value['enum'] as Set).cast<String>();
```

Comment:
This is typically going to be a `List`, not a `Set`. We could call `toSet`, but I would probably just make it have an `Iterable` type so that it can be either? We should also validate that the JSON encoder can handle any kind of Iterable.